app/src/keymap.c

```diff
@@ -115,8 +121,20 @@ int zmk_keymap_layer_toggle(uint8_t layer) {
     return zmk_keymap_layer_activate(layer);
 };
 
-bool is_active_layer(uint8_t layer, zmk_keymap_layers_state layer_state) {
-    return (layer_state & BIT(layer)) == BIT(layer) || layer == _zmk_keymap_layer_default;
+int zmk_keymap_layer_to(uint8_t layer) {
```

Comment:
this method causes 32 layer_state_changed events (31 for disabling all layers except the default layer + 1 for enabling the to-layer).

Layer-changed-events should only be triggered if the layer state actually changes.

Not sure if the layer that is being moved to should be disabled & enabled again (or just kept enabled if it was already enabled). I don't have a strong opinion on this.